lxd/storage/drivers/driver_lvm_utils.go

```diff
@@ -501,11 +508,42 @@ func (d *lvm) lvmDevPath(vgName string, volType VolumeType, contentType ContentT
 
 // resizeLogicalVolume resizes an LVM logical volume. This function does not resize any filesystem inside the LV.
 func (d *lvm) resizeLogicalVolume(lvPath string, sizeBytes int64) error {
-	_, err := shared.TryRunCommand("lvresize", "-L", fmt.Sprintf("%db", sizeBytes), "-f", lvPath)
+	isRecent, err := d.lvmVersionIsAtLeast(lvmVersion, "2.03.17")
+	if err != nil {
+		return fmt.Errorf("Error checking LVM version: %w", err)
+	}
+
+	args := []string{"-L", strconv.FormatInt(sizeBytes, 10) + "b", "-f", lvPath}
+	if isRecent {
+		args = append(args, "--fs=ignore")
+	}
+
+	// If needed, get the original LV size.
+	var oldSizeBytes int64
+	needsClearing := !d.usesThinpool()
+	if needsClearing {
+		oldSizeBytes, err = d.logicalVolumeSize(lvPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = shared.TryRunCommand("lvresize", args...)
 	if err != nil {
 		return err
 	}
 
+	// On thick pools, discard the blocks in the additional space when the volume is grown.
+	if needsClearing {
+		if oldSizeBytes < sizeBytes {
+			// Discard the new blocks.
+			err := block.ClearBlock(lvPath, sizeBytes)
```

Comment:
At that point, the LV has already been grown so shouldn't we call `ClearBlock` with `oldSizeBytes` so that it zero's from `oldSizeBytes` up to `sizeBytes`?

Reply:
yes i think you could be right here, will take a look thanks

Reply:
thanks, ive reworked the thick volume grown code as it had additional problems.

It didn't work basically, because the lvm volume wasn't activated when ClearBlock was called and the ClearBlock function silently ignored that. I strongly like that kind of "magic" behaviour, so instead of having ClearBlock ignore missing files, its now up to the caller to decide if that is a valid error to ignore or not.

Reply:
Have now confirmed that creating a block volume at 10MiB, filling it with "A"s, counting them and then growing the volume, checking that it contained the same amount of As as before and no other data.